src/services/utilities.ts

```diff
@@ -771,6 +771,20 @@ namespace ts {
         }
     }
 
+    function getAncestorTypeNode(node: Node) {
+        return findAncestor(node, a => !isQualifiedName(a.parent) && !isTypeNode(a.parent) && !isTypeElement(a.parent));

```

Comment:
This only works if the input node is actually inside a type node, right? If you call it on a random expression node, it will just return that node?